### PR TITLE
Etr ws

### PR DESCRIPTION
**Pull request by mdovgialo**

### CODE REVIEW

File: obci/drivers/etr/etr_amplifier_ws_peer.py [hunk 1]
```diff
@@ -8,61 +8,105 @@
 
 from obci.drivers.etr import etr_amplifier
 from obci.configs import settings, variables_pb2
+from multiplexer.multiplexer_constants import peers, types
 
 from ws4py.client.threadedclient import WebSocketClient
 
 class EtrWebSocketClient(WebSocketClient):
```
Comment (mdovgialo) [inline obci/drivers/etr/etr_amplifier_ws_peer.py]: not needed

File: obci/drivers/etr/etr_amplifier_ws_peer.py [hunk 2]
```diff
@@ -8,61 +8,105 @@
...
+    Modification to EtrWebSocketClient, which lets to capture more data
+    than x and y gaze position.
+    '''
+    def __init__(self, url, peer,  packet = 4):
+        super(EtrWebSocketClientSample, self).__init__(url, peer,)
         
+        self.packetsize = packet
+        self.msg_vector = variables_pb2.SampleVector()
+    def received_message(self, m):
+
```
Comment (mdovgialo) [inline obci/drivers/etr/etr_amplifier_ws_peer.py]: check if requested channels are available in json

raise BIG error

File: obci/drivers/etr/etr_amplifier_ws_peer.py [hunk 3]
```diff
@@ -8,61 +8,105 @@
...
+    def received_message(self, m):
+            
+        msg = json.loads(str(m))
+        #~ self.peer.logger.info("msg: {}".format(m))
+        timestamp = float(msg['timestamp'])
+        sample = self.msg_vector.samples.add()
+        sample.timestamp = timestamp
+        
+        sample.channels.extend(
+            [float(msg['{}'.format(i)]) for i in self.peer.chs]
```
Comment (mdovgialo) [inline obci/drivers/etr/etr_amplifier_ws_peer.py]: msg[i]

Comment (mdovgialo) [inline obci/drivers/etr/etr_amplifier_ws_peer.py]: ( < [

File: obci/drivers/etr/etr_amplifier_ws_peer.py [hunk 4]
```diff
@@ -8,61 +8,105 @@
...
             str(self.config.get_param('etr_ws_server_host')),
             str(self.config.get_param('etr_ws_server_port'))
         )
-        
-        self.ws = EtrWebSocketClient(url)
+        message_type = str(self.config.get_param('message_type'))
+        if message_type=='sample2d':
+            self.ws = EtrWebSocketClient(url, self)
+        elif message_type == 'samplevector':
+            packetsize = int(self.config.get_param('samples_per_packet'))
```
Comment (mdovgialo) [inline obci/drivers/etr/etr_amplifier_ws_peer.py]: change packetsize to samples_per_packet

File: obci/drivers/etr/etr_amplifier_ws_peer.py [hunk 5]
```diff
@@ -8,61 +8,105 @@
...
+        super(EtrWebSocketAmplifier, self).__init__(addresses)
         url = 'ws://{}:{}/ws'.format(
             str(self.config.get_param('etr_ws_server_host')),
             str(self.config.get_param('etr_ws_server_port'))
         )
-        
-        self.ws = EtrWebSocketClient(url)
+        message_type = str(self.config.get_param('message_type'))
+        if message_type=='sample2d':
+            self.ws = EtrWebSocketClient(url, self)
```
Comment (mdovgialo) [inline obci/drivers/etr/etr_amplifier_ws_peer.py]: remove legacy

File: obci/drivers/etr/etr_amplifier_ws_peer.py [hunk 6]
```diff
@@ -8,61 +8,105 @@
...
+            packetsize = int(self.config.get_param('samples_per_packet'))
+            self.ws = EtrWebSocketClientSample(url, self, packetsize)
+            self.chs = self.get_param("channel_names").split(';')
+            self.set_param('channel_gains', ';'.join(
+                                        [str(1.0) for i in self.chs]))
+            self.set_param('channel_offsets', ';'.join(
+                                          [str(0.0) for i in self.chs]))
+            self.number_of_channels = int(len(self.chs))
+        else:
+            raise(Exception('Unrecognised option: message_type'))
```
Comment (mdovgialo) [inline obci/drivers/etr/etr_amplifier_ws_peer.py]: raise

File: obci/drivers/etr/etr_amplifier_ws_peer.py [hunk 7]
```diff
@@ -8,61 +8,105 @@
...
+            
+            
         self.ws.daemon = False
         self.ws.connect()
         
         self.end_the_loop = False
```
Comment (mdovgialo) [inline obci/drivers/etr/etr_amplifier_ws_peer.py]: logger.info

File: obci/drivers/etr/etr_amplifier_ws_peer.py (hunk 2 above)
Comment (mdovgialo) [inline obci/drivers/etr/etr_amplifier_ws_peer.py]: like:
2016-04-12 17:09:59,587 - etr_amplifier - INFO - Opened
2016-04-12 17:09:59,609 - etr_amplifier - ERROR - 
            
            ERROR: UNAVAILABLE CHANNELS:
            ooo
            
            
            Channels from tracker server JSON:
            y, timestamp, eye_detected, x
            
            You've tried to access channels:
            
            timestamp, eye_detected, x, y, ooo
            
            
            
            
2016-04-12 17:09:59,609 - etr_amplifier - INFO - ('Closed down', 1006, 'Going away')

**Comment by mdovgialo** [comment]
@mroja 
@develancer 
@achabuda 
Done

File: obci/drivers/etr/etr_amplifier_ws_peer.py [hunk 8]
```diff
@@ -8,52 +8,111 @@
...
-            l_msg.y = float(m['y'])
-            l_msg.timestamp = float(m['timestamp'])
+            Channels from tracker server JSON:
+            {}
             
-            self.peer.process_message(l_msg)
-        except Exception as ex:
-            pass
 
+            You've tried to access channels:
```
Comment (mroja) [inline obci/drivers/etr/etr_amplifier_ws_peer.py]: This looks **very** unreadable. Use named variables in format.

Comment (mdovgialo) [inline obci/drivers/etr/etr_amplifier_ws_peer.py]: Now should be better

**Pull request merged**